torch_geometric/explain/explainer.py

```diff
@@ -128,6 +133,8 @@ def __call__(
             **kwargs,
         )
 
+        self.model.train(training)
```

Comment:
Add a test to check if model state is reset. This will prevent someone from modifying this by mistake in the future.